### ISSUE TITLE
[HW2] Backend or InMemoryDataService

### ISSUE THREAD

**Issue by SoheeLee** [swsnu/swppfall2017#40]
Backend를 만들지 않고 skeleton code에 구현된 InMemoryDataService를 사용하면 되나요?

Should we use implemented InMemoryDataService instead of implementing backend?

**Comment by wonook** [comment]
Yes, that is exactly the intention of the skeleton code. You are not expected to create any backend code

**Comment by SoheeLee** [comment]
빠른 답변 감사합니다!
InMemoryDataService로 구현된 DB를 사용해보았는데, `update`를 구현하여 `/sign_in` page에서 sign in 했을 때 `user 1` (swpp@snu.ac.kr)의 `signed_in` flag를 true로 update할 수 있었습니다. 그러나 다음과 같은 현상이 일어났습니다.
- sign in한 상태(`/articles` page)에서 page를 refresh했을 때  `user 1`의 `signed_in` flag가 `false`로 초기화되어, 자동으로 sign out되는 현상이 일어났습니다.
- 또한 sign in한 상태(`/articles` page)에서 url을 통해 main page(`localhost:4200`)에 접근했을 때도 user 1의 `signed_in` flag가 `false`로 초기화되었습니다.

제 구현에 문제가 있었던게 아닐까 해서 똑같이 `InMemoryDbService` module을 사용하는`https://angular.io/tutorial/toh-pt6`의 example을 다운받아 테스트해보았는데, 이 example에서도 위의 현상과 유사하게
- `http://localhost:3000/detail/:id`에서 Hero의 detail을 수정해 `/dashboard` page로 돌아온 후 url을 통해 main page(`localhost:3000`)에 접근했을 때 모든 수정내역이 초기화되었습니다.
- page를 refresh했을 때도 모든 수정내역이 초기화되었습니다.

그렇다면 page를 refresh하거나, url을 통해 이동했을 때 수정내역이 초기화되는 문제는 고려하지 않아도 되는 것인가요? 스펙에 제시된 대로 sign in하지 않은 상태에서 url을 통해 `/articles` page에 접근할 수 없도록만 만들면, 그 외의 url을 통해 이동하는 경우는 고려하지 않아도 되나요?

**Comment by wonook** [comment]
Yes, as we don't have a proper backend, we don't consider the cases where users refresh their pages just yet. (but such cases should work when we have a proper backend with hw3!). It will be fine as long as it treats such cases as a user not signed in accessing the application, which should redirect the anonymous user to the sign_in page. The `Sign In` page should be the only page that unauthorized users can have access to. 😃
